.golangci.yml

```diff
@@ -12,7 +12,6 @@ linters:
   - gci
   - goconst
   - gocritic
-  - gocyclo
```

Comment:
This seems unrelated to the goal of this PR 🤔

Reply:
This PR will hit the gocyclo limit on the kcp controllers reconcile function. This was done to address that. Ref discussion here: https://github.com/kubernetes-sigs/cluster-api/pull/6183#discussion_r812134821

Shall I drop the gocyclo linter in a separate PR and then rebase this one?

Reply:
Separate PR sounds good to me

Reply:
PR: https://github.com/kubernetes-sigs/cluster-api/pull/6193